### DISCUSSION TITLE
Detecting small objects in yolov8x-pose

### DISCUSSION THREAD

**Discussion by AdwaitMD** [ultralytics/ultralytics#10459] [Q&A]
Hi, 
I am training a model to detect small objects with  "yolov8x-pose". I need to detect and then track their body posture. To give you an idea,  My raw image resolution is 5K, and the smallest object are 80*20 px.  I am training the models at 1280 and also at inference using 1280 as imgsz. Also, I have augmented. Iwas wondering other than increasing training data, are there any tips and suggestions for increasing performance of the model. Is there any equivalent of "YOLOv8m-p2'  for pose models geared towards small object detection, or can I change something specific at the layers level ? or some parameters while training?  
Thanks.

**Comment by pderrenger** [comment]
@AdwaitMD hi there! 😊

For improving detection and pose estimation of small objects with `yolov8x-pose`, you're on the right track by training at a higher image size and using augmentation. Here are a few more tips that might help:

- **Focus on Anchor Sizes:** Adjust the anchor sizes to better match the scale of your small objects. Custom anchors more representative of your object sizes can be more effective.

- **Model Customization:** Consider customizing the model by adjusting layers or adding attention mechanisms to focus on smaller object features. Modifying the depth, width, or resolutions through the model’s configuration file could also help.

- **Use Higher Resolution:** Continue using high resolution (`imgsz=1280`) for both training and inference, as this retains more details for small objects.

- **Advanced Augmentations:** Experiment with advanced augmentation strategies like cutout, mixup, or mosaic to provide diverse examples, especially focusing on scenarios with small-sized objects.

- **Tiling Strategy:** For 5K resolution images, consider using a tiling strategy during both training and inference. Divide your images into smaller tiles (ensuring overlap), run detection on each, and consolidate the results. This could allow the model to "see" small objects with more context.

- **Fine-tuning Parameters:** Experiment with training hyperparameters, especially focus on lowering `obj_loss_gain` and possibly adjusting `iou_loss_gain` if you notice the model struggling with precise localization or distinguishing close objects.

There isn't a direct equivalent of "YOLOv8m-p2" for pose models specifically designed for small object detection in the current Ultralytics arsenal. However, customizing your model by modifying `yolov8x-pose` architecture directly for your specific use case, as above, can often bridge this gap.

Remember, small tweaks can lead to significant improvements, don't hesitate to experiment. Keep up the great work!

```bash
# Example anchor tuning command
yolo train --data your_data.yaml --img 1280 --batch 16 --epochs 300 --cfg yolov8x-pose.yaml --weights '' --name your_model_name
```

Best of luck with your model training! 👍
